tcms/testplans/templates/testplans/mutable.html

```diff
@@ -0,0 +1,96 @@
+{% extends "base.html" %}
+{% load i18n %}
+{% load static %}
+
+{% block head %}
+    {{ form.media }}
+{% endblock %}
+
+<head>
+    <title>{% block title %}{% trans "Create new TestPlan" %}{% endblock %}</title>
+</head>
+
+{% block contents %}
+
+    <div class="container-fluid container-cards-pf">
+        <form class="form-horizontal" action="{% url 'plans-new' %}" enctype="multipart/form-data" method="post">
+            {% csrf_token %}
+
+            <div class="form-group">
+                <label class="col-md-1 col-lg-1" for="id_name">{% trans "Name" %}</label>
+                <div class="col-md-11 col-lg-11">
+                    <input type="text" id="id_name" name="name" value="{% firstof form.name.value "" %}" class="form-control">
+                </div>
+            </div>
+
+            <div class="form-group">
+                <div class="col-md-1 col-lg-1">
+                    <label for="id_product">{% trans "Product" %}</label>
+                    <a href="{% url 'admin:management_product_add' %}?_popup" id="add_id_product" alt="{% trans 'add new Product' %}" title="{% trans 'add new Product' %}">+</a>
+                </div>
+                <div class="col-md-3 col-lg-3">
+                    <select id="id_product" name="product" class="form-control">
+                        {% for product in form.product.field.queryset %}
+                            <option value="{{ product.pk }} {% if product.pk == form.product.value %}selected{% endif %}">
+                                {{ product.name }}
+                            </option>
+                        {% endfor %}
+                    </select>
+                </div>
+                <div class="col-md-1 col-lg-1">
+                    <label for="id_version">{% trans "Version" %}</label>
+                    <a href="{% url 'admin:management_version_add' %}?_popup"
```

Comment:
This link will open the popup form to add new Version without the Product field being pre-selected. So it is possible for the user to create a Product version against a different product, not the one currently being selected.

So your on-product-change handler needs to append to this URL `&product=X` where `X` is the ID of the actual product selected from the Product drop down widget. In this way when the user clicks on the + button the popup that opens will have the same product pre-selected.